### ISSUE TITLE
catalog-operator in CrashLoopBackOff

### ISSUE THREAD

**Issue by rohasharma** [operator-framework/operator-lifecycle-manager#2872]
catalog-operator Pod is in CrashLoopBackOff state post deploying OLM using helm command. 
The reason shown in the logs are:
Error: unknown flag: --set-workload-user-id
unknown flag: --set-workload-user-id

Irrespective if we set the flag in values as True or False same behaviour is observed. The only way to proceed right now is to remove the else statement from 0000_50_olm_08-catalog-operator.deployment.yaml and then set setWorkloadUserID: true as false in values.yaml.

My kubernetes is running v1.23.10.

Thanks
Rohan Roy Sharma

**Comment by dmesser** [comment]
I think this may be due to the fact that the chart default value for the catalog operator image points to quay.io/operator-framework/olm:master - this tag hasn't been updated since the last 6 months. Your workaround would be to use any of the newer tags, e.g. quay.io/operator-framework/olm:v0.22

@anik120 The build job for OLM that generates an image according to branch name on quay.io is disabled. I guess you may want to either re-enable or make the helm chart part of the upstream release so it gets a versioned image.

**Comment by rohasharma** [comment]
Thank you so much @dmesser for the response. I was in-fact hardcoding the version from master to v0.22 in past but along with it I did remove the flag from values.yaml as a workaround. Since we are automating the OLM deployment using flux we want to make least possible change in the git repo so that we can import it as it is. 

Since we are on this topic of using default templates/values from this repo in our environment there is one more thing I want to bring in notice here: 
These template files has flag set for runAsNonRoot: true
`chart/templates/0000_50_olm_07-olm-operator.deployment.yaml
chart/templates/_packageserver.deployment-spec.yaml
chart/templates/0000_50_olm_08-catalog-operator.deployment.yaml`
However, if we run the helm install as it is, it fails the deployment and as a work around we need to add 
`runAsUser: 1000`
Should we add it in our template files so that we don't see any failure with pod deployments?

**Comment by anik120** [comment]
[1001](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/cmd/catalog/main.go#L33) is [already being set as the default ID](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/cmd/catalog/main.go#L87-L90), so updating that image will take care of that too (I.e the reason it's erroring out for you is coz the latest olm isn't actually running. So you won't need to add `runAsUser:1000` to your template files with the latest olm). Looking into the updating of the `quay.io/operator-framework/olm:master` image now.

**Comment by rohasharma** [comment]
> [1001](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/cmd/catalog/main.go#L33) is [already being set as the default ID](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/cmd/catalog/main.go#L87-L90), so updating that image will take care of that too (I.e the reason it's erroring out for you is coz the latest olm isn't actually running. So you won't need to add `runAsUser:1000` to your template files with the latest olm). Looking into the updating of the `quay.io/operator-framework/olm:master` image now.

Hi @anik120, thanks for the response. 
Should I be expecting a new tag like v0.22? 
I can't use olm:master in my values as we are planning to use the olm repository  in production to deploy our olm. 
Using master can break stuff for us with future commits.

**Comment by anik120** [comment]
@rohasharma we just realized that we haven't been releasing manifests for OLM release after v0.18.3 [here](https://github.com/operator-framework/operator-lifecycle-manager/tree/master/deploy/upstream/manifests). That was an oversight from our team and we apologize for the confusion. I've started the process of releasing the v0.22.0 manifests with this PR: https://github.com/operator-framework/operator-lifecycle-manager/pull/2873

That will be followed by another PR that will add the deploy/upstream/manifests/v0.22.0 folder which will contain the v0.22.0 manifests, that you'll be able to use. I'll update here once that is done. 


FYI @dmesser I've also re-enabled the triggered job for `quay.io/operator-framework/olm:master` image so folks deploying using the Quickstart will not see a broken installation anymore.

**Comment by anik120** [comment]
@rohasharma is there a reason why you're using the helm chart instead of using the release artifacts from the [v0.22.0 release](https://github.com/operator-framework/operator-lifecycle-manager/releases/tag/v0.22.0)? Could you use the release artifacts to deploy OLM instead?

**Comment by rohasharma** [comment]
Hi @anik120, yes the reason why we are using Helm is to integrate it with FluxCD for automatic deployment of OLM namespace with pods running inside them.

**Comment by anik120** [comment]
@rohasharma could you shed more light who your group is? And how exactly OLM is being deployed (a link to the code would be useful)

**Comment by perdasilva** [comment]
I'm closing the issue and the PR due to inactivity. Please re-open if necessary ^^
